README.md

```diff
@@ -210,9 +213,125 @@ func AsynchronousFriends(forUser: jimbob) -> Deferred<[Friend]> {
 }
 ```
 
+## Usage - Producer
+
+```swift
+// Potentially long-running operation.
+func performOperation() -> Deferred<Int> {
+		// 1. Create deferred.
+    let deferred = Deferred<Int>()
+
+		// 2. Kick off asynchronous code that will eventually...
+    dispatch_async(dispatch_get_main_queue(), {
+        let result = compute_result()
+
+				// 3. ... fill the deferred in with its value
+        deferred.fill(result)
+    })
+
+		// 4. Return the (currently still unfilled) deferred
+    return deferred
+}
+```
+
+## Usage - Consumer
+
+### <a name="upon"></a>Running Closures Upon Fulfillment
+
+You can use the `upon` method to run a closure once the `Deferred` has been
+filled. `upon` can be called multiple times, and the closures will be called
+in the order they were supplied to `upon` (with the normal race condition caveat
+if you are calling `upon` from multiple threads simultaneously).
+
+By default, `upon` will run the closures on a background concurrent GCD queue.
+You can change this by passing a different default queue when the `Deferred` is
+created, or by using the `uponQueue` method to specify a queue for the closure.
+
+```swift
+let deferredResult = performOperation()
+
+deferredResult.upon { result in
+    println("got \(result)")
+}
+```
+
+### Peeking at Current Value
+
+Use the `peek` method to determine whether or not the `Deferred` is currently
+filled.
+
+```swift
+let deferredResult = performOperation()
+
+if let result = deferredResult.peek() {
+		println("filled with \(result)")
+} else {
+		println("currently unfilled")
+}
+```
+
+### Blocking on Fulfillment
+
+Use the `value` property to wait for the `Deferred` to be filled and get the value.
+
+```swift
+// WARNING: Blocks the calling thread!
+let result: Int = performOperation().value
```

Comment:
We removed `.value` as a public API. We could demonstrate the use of `wait(_:)` instead.